### PR TITLE
Cleanup documentation

### PR DESCRIPTION
**Pull request by resourcefulmind**
# Comprehensive Cleanup and Documentation Reorganization for Akeru

## Description
This pull request implements a series of enhancements aimed at cleaning up and better organizing the documentation and configuration files within the Akeru repository. These changes are designed to improve navigation, enhance readability, and align with best practices for project maintenance.

## Key Changes
- **Documentation Reorganization**: Moved all relevant documentation files into a `docs` folder and updated file extensions to lowercase to maintain consistency across the project.
- **Issue and PR Templates Relocation**: Restructured where we store our issue and pull request templates to ensure they are utilized effectively by GitHub’s automation and are easier for contributors to find.
- **Removal of Unused Lock Files**: Deleted `package-lock.json` and updated `.gitignore` to prevent it from being tracked, aligning with our use of Bun as our package manager.
- **Documentation Updates**: Made updates to internal references in documentation to reflect new file locations and ensure all links are functional.

### Detailed Changes
- Moved `CODE_OF_CONDUCT.md`, `CODING_CONVENTIONS.md`, `CONTRIBUTING.md`, and `DOCUMENTATION_STYLE_GUIDE.md` to the `docs` directory.
- Renamed `CODING_CONVENTIONS.MD` to `coding_conventions.md` to standardize naming conventions.
- Transferred `ISSUES_TEMPLATE.md` to `.github/ISSUE_TEMPLATE` and `PULL_REQUEST_DOC_TEMPLATE.md` to `.github/PULL_REQUEST_DOC_TEMPLATE`, assessing its usage versus the existing `PULL_REQUEST_TEMPLATE.md`.
- Added `package-lock.json` to `.gitignore` and removed existing file from the repository.

## Motivation and Context
These updates address feedback from the community and internal discussions about maintaining a clean and navigable repository. By organizing our documentation and templates more intuitively and removing unused files, we enhance every contributor's ability to interact with and contribute to the project efficiently.

## Link to Issue
- Addresses feedback and suggestions from issue #110 .

## Request for Review
- @GuiBibeau  - For review of documentation structure and updates.
- @GuiBibeau  - For verification of template functionality and file deletions.

Looking forward to your feedback and suggestions on these updates. Your insights are invaluable to ensure that our repository remains well-organized and our documentation is clear and accessible.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -180,4 +180,4 @@ dist
 .env.development
 .env.test
 
-.DS_Store
\ No newline at end of file
+.DS_Storepackage-lock.json
```
Comment (GuiBibeau) [inline .gitignore:183]: let's add each on a split line and not on the same line

**Review by GuiBibeau — COMMENTED** [review]
one change needed only!

File: .gitignore (hunk 1 above)
Comment (resourcefulmind) [reply .gitignore]: Sure thing

**Review by GuiBibeau — APPROVED** [review]
beautiful thank you

**Pull request merged**